lib/src/util/dart_type_utilities.dart

```diff
@@ -47,13 +47,33 @@ class DartTypeUtilities {
   static bool hasInheritedMethod(MethodDeclaration node) =>
       lookUpInheritedMethod(node) != null;
 
+  static InterfaceType implementedInterface(
+      DartType type, String interface, String library) {
+    if (type is! InterfaceType) {
+      return null;
+    }
+    bool predicate(InterfaceType i) => isInterface(i, interface, library);
+
+    ClassElement element = type.element;
+    if (predicate(type)) {
+      return type;
+    }
+    if (!element.isSynthetic && type is InterfaceType) {
+      final interfaces = element.allSupertypes.where(predicate);
```

Comment:
You are totally right! the problem was that I was using interfaceType.element.allSupertypes instead interfaceType.superclass and interfaceType.interfaces so I lost the propagated argumentList but I can rescue now, thank you!